### PR TITLE
Solidity plonk verifier

### PR DESCRIPTION
**Pull request by lazovicff**

### CODE REVIEW

File: circuit/src/utils.rs [hunk 1]
```diff
@@ -26,7 +26,104 @@ use halo2::{
...
+	let mut buffer = String::new();
+
+	// Read file into string.
+	reader.read_to_string(&mut buffer).unwrap();
+
+	buffer
+}
+
+/// Reads raw bytes from the file
+pub fn read_bytes_data(name: &str) -> Vec<u8> {
```
Comment (duguorong009) [inline circuit/src/utils.rs:64]: How about `bin_file_name` or `file_name`, instead of `name`?

File: circuit/src/utils.rs [hunk 2]
```diff
@@ -26,7 +26,104 @@ use halo2::{
...
+	read_bytes(bin_path)
+}
+
+/// Write bytes to a file
+pub fn write_bytes(bytes: Vec<u8>, path: impl AsRef<Path>) -> Result<(), IoError> {
+	write(path, bytes)
+}
+
+/// Write bytes to data directory
+pub fn write_bytes_data(bytes: Vec<u8>, name: &str) -> Result<(), IoError> {
```
Comment (duguorong009) [inline circuit/src/utils.rs:76]: Same as above.
I think it is better to use clear variable name, like `file_name`, instead of just `name`.

**Review by duguorong009 — COMMENTED** [review]
I left some comments.
Simply, I think it is better to give clear variable names to utility functions in `circuit/src/utils.rs`.

File: circuit/src/utils.rs (hunk 1 above)
Comment (lazovicff) [reply circuit/src/utils.rs:64]: Sure, we can do that

**Pull request merged**